src/app/demo-page/demo-page.component.ts

```
@@ -22,31 +23,35 @@ export class DemoPageComponent{
   ngOnInit() {
   }
 
-  public callTestEndpoint(): Promise<boolean> {
-    return new Promise(resolve => {
-      this.demoPageService.callTestEndpoint()
-      .then((result) => {
-        console.log(result);
-        this.body = result;
-        resolve(true);
-      })
-      .catch((error) => {
-        resolve(false);
-      });
-    });
+  public callTestEndpoint(): Observable<boolean> {
+	var obs = this.demoPageService.callTestEndpoint()
+				 .map(result =>
+				 {
+					this.body = (result == undefined) ? undefined : result.data;
+					return !(this.body === undefined);
+                 });
+	   
+	   obs.subscribe();
+	   
+	   return obs;
   }
 
-  public callTestEndpointPopulateObject(): Promise<boolean> {
-    return new Promise(resolve => {
-      this.demoPageService.callTestEndpointPopulateObject()
-      .then((result) => {
-        console.log(result);
-        this.responseObject = result;
-        this.mind = this.responseObject.data;
-      })
-      .catch((error) => {
-        resolve(false);
-      });
-    });
+  public callTestEndpointPopulateObject(): Observable<boolean> {	  
+	  var obs = this.demoPageService.callTestEndpointPopulateObject()
+				 .map(result =>
+				 {
+					 if(result == null || result == undefined) {
```

I would change this to 

if(result)
...
else return false;

If _result_ is undefined or null then: _if(result)_ will return false.

---

Plus. We should use the ternary here, too.

---

Or we can just let it throw an error if the data is missing. But then the tests will break. :)

---

Maybe we don't need a couple of those tests... that would help us get rid of the Observable<boolean> stuff.